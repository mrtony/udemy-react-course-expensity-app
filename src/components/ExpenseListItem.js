import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = props => (
    <div>
        <Link to={`/edit/${props.id}`}><h3>Description: {props.description}</h3></Link>
        <p>Note: {props.amount}-{props.createdAt}</p>
    </div>
);

export default ExpenseListItem;