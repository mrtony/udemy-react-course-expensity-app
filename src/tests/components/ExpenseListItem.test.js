import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';

const expense = {
  description: 'Gum',
  note: 'my note',
  amount: 195,
  createdAt: moment(0).format('YYYY/MM/DD')
};

describe('ExpenseListItem component', () => {
  test('should ExpenseListItem render without expense correctly', () => {
    const wrapper = shallow(<ExpenseListItem />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should ExpenseListItem render with id and expense correctly', () => {
    const wrapper = shallow(<ExpenseListItem id="1" {...expense} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Route to create page should correctly', () => {
    const wrapper = shallow(<ExpenseListItem id="1" />);
    const linkTo = wrapper.find('Link').prop('to');
    expect(linkTo).toBe('/edit/1');
  });
})