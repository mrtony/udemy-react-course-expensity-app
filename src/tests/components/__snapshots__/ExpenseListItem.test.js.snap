// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseListItem component should ExpenseListItem render with id and expense correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Description: 
      Gum
    </h3>
  </Link>
  <p>
    Note:
    $1.95
    -
    January 1st 1970
  </p>
</div>
`;

exports[`ExpenseListItem component should ExpenseListItem render without expense correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/undefined"
  >
    <h3>
      Description: 
    </h3>
  </Link>
  <p>
    Note:
    $0.00
    -
    January 1st 1970
  </p>
</div>
`;
